src/v/kafka/protocol/schemata/generator.py

```diff
@@ -926,6 +929,33 @@ def _redpanda_decoder(self):
 
         raise Exception(f"No decoder for {(tn, fn)}")
 
+    def _redpanda_validator(self):
+        """
+        Checks the type to see if there is a validator function call
+        :return: The name of the function to call to validate or None
+        """
+        path_type = self._redpanda_path_type()
+        if path_type:
+            return None if path_type[
+                1] not in validator_function_map else validator_function_map[
+                    path_type[1]]
+
+        et = self._field.get("entityType", None)
+        if et in entity_type_map:
+            m = entity_type_map[et]
+            return None if m[
+                1] not in validator_function_map else validator_function_map[
+                    m[1]]
+
+        tn = self._type.name
+        fn = self._field["name"]
+
+        if (tn, fn) in field_name_type_map or tn in basic_type_map:
+            return None if tn not in validator_function_map else validator_function_map[
+                tn]
+
+        return None
+
```

Comment:
i like this idea of having a validator, but it's not clear why we need to code generate it. it seems like we could add a call to the control character validator in `request_reader.h`?

Reply:
Yeah i'd agree with that. Our generator is already complex and messy, its hard to extend and its probably better to avoid changing it if its at all necessary.